pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -69,20 +98,41 @@ func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTime
 				}
 			}
 		}()
+
+		go func() {
+			for {
+				select {
+				case <-p.close:
+					return
+				// Persist lazy loaded blocks in file every one minute
+				case <-time.After(1 * time.Minute):
+					// We copy the state of blocks that are being lazy loaded from LazyBinaryReaders.
```

Comment:
nit: but this comment doesn't say more than the line below. I'm tempted to remove it. Or maybe give some explanation for why we do the copy and persist as two operations (to not hold the lock for long)

Reply:
Make sense. I will update the comment.